### PR TITLE
[Bugfix] use Model::$cacheQueries as default for ->query(...) (#9575)

### PR DESCRIPTION
**Pull request by dmaicher**
This is my attempt of fixing https://github.com/cakephp/cakephp/issues/9575.

Note: I never worked with cakephp before :see_no_evil:

### CODE REVIEW

File: lib/Cake/Test/Case/Model/ModelReadTest.php [hunk 1]
```diff
@@ -8531,4 +8531,48 @@ public function testRelatedAfterFindCallback() {
...
+				$query = sprintf('SELECT u.user FROM %s u ORDER BY id DESC LIMIT 1', $userTableName);
+
+				if ($cacheArgument !== null) {
+					$users = $model->query($query, $cacheArgument);
+				} else {
+					$users = $model->query($query);
+				}
+			}
+
+			return isset($users[0][0]) ? $users[0][0]['user'] : $users[0]['u']['user'];
```
Comment (markstory) [inline lib/Cake/Test/Case/Model/ModelReadTest.php]: This is a pretty complex piece of test code. Could we simplify it by breaking down each of the cases into separate test methods? I find test code is easier to understand when the logic is simple, and there is some duplication.

**Comment by markstory** [comment]
Your [tests are failing on postgres](https://travis-ci.org/cakephp/cakephp/jobs/166114042)

File: lib/Cake/Test/Case/Model/ModelReadTest.php (hunk 1 above)
Comment (dmaicher) [inline lib/Cake/Test/Case/Model/ModelReadTest.php]: I broke it down into 4 separate test cases now :wink:

**Comment by dmaicher** [comment]
Tests are passing now as well :wink:

**Comment by markstory** [comment]
Thanks :+1:

**Pull request merged**